### PR TITLE
COMP: add completion for the non_exhaustive attribute

### PR DESCRIPTION
**Pull request by Kobzol**
Fixes: https://github.com/intellij-rust/intellij-rust/issues/6334

### CODE REVIEW

**Review by Undin — COMMENTED** [review]
@Kobzol Could you rebase this PR onto master to resolve conflicts?

**Comment by Kobzol** [comment]
More conflicts appeared, rebased again.

File: src/test/kotlin/org/rust/lang/core/completion/RsAttributeCompletionTest.kt [hunk 1]
```diff
@@ -221,4 +221,24 @@ class RsAttributeCompletionTest : RsCompletionTestBase() {
...
+
+    fun `test non_exhaustive on enum`() = doSingleCompletion("""
+        #[non_/*caret*/]
+        enum Foo {}
+    """, """
+        #[non_exhaustive/*caret*/]
+        enum Foo {}
+    """)
+
+    fun `test non_exhaustive on struct`() = doSingleCompletion("""
```
Comment (Undin) [inline src/test/kotlin/org/rust/lang/core/completion/RsAttributeCompletionTest.kt:233]: Let's add two more tests: for enum variant and for some item (e.g. function) where this attribute shouldn't be completed

**Review by Undin — APPROVED** [review]
Thank you!

**Comment by Kobzol** [comment]
bors r=undin

**Pull request merged**